XenModel/Actions/Updates/DownloadAndUpdateClientAction.cs

```diff
@@ -102,6 +105,13 @@ private void DownloadFile()
 
                     //start the download
                     _updateDownloadState = DownloadState.InProgress;
+
+                    if (!string.IsNullOrEmpty(_authToken))
+                    {
+                        NameValueCollection myQueryStringCollection = new NameValueCollection();
+                        myQueryStringCollection.Add("InternalStageAuthToken", _authToken);

```

Comment:
You may want to use the constant from the Registry.cs instead of hardcoding, so in future you'll have to change it only in one place.

Reply:
Ah thats a good idea